### PR TITLE
Add support for 0xefa1 device

### PR DESCRIPTION
**Pull request by galpress**
Kernel submission:
v1: https://lore.kernel.org/linux-rdma/20200709083630.21377-1-galpress@amazon.com/
v2: https://lore.kernel.org/linux-rdma/20200720080113.13055-1-galpress@amazon.com/
v3: https://lore.kernel.org/linux-rdma/20200721133049.74349-1-galpress@amazon.com/
v4: https://lore.kernel.org/linux-rdma/20200722140312.3651-1-galpress@amazon.com/

### CODE REVIEW

File: providers/efa/efa.c [hunk 1]
```diff
@@ -20,6 +20,7 @@ static void efa_free_context(struct ibv_context *ibvctx);
 static const struct verbs_match_ent efa_table[] = {
 	VERBS_DRIVER_ID(RDMA_DRIVER_EFA),
 	VERBS_PCI_MATCH(PCI_VENDOR_ID_AMAZON, 0xefa0, NULL),
+	VERBS_PCI_MATCH(PCI_VENDOR_ID_AMAZON, 0xefa1, NULL),
```
Comment (jgunthorpe) [inline providers/efa/efa.c:23]: No more updates to pci driver tables. VERBS_DRIVER_ID covers everything.

**Comment by jgunthorpe** [comment]
Keep in mind the provider today will load on the new driver. Is it Ok or do you need more compatability negotaition during alloc_ucontext?

**Comment by galpress** [comment]
> Keep in mind the provider today will load on the new driver. Is it Ok or do you need more compatability negotaition during alloc_ucontext?

No, that's not going to work with today's provider.
What's the motivation to stop using pci ids? Isn't preventing the provider to load on this id better than a ucontext handshake?

**Comment by jgunthorpe** [comment]
All the providers are expected to negotiate with the kernel if the right combination stuff is supported.

Will this upgraded provider work on old kernels? That would be the rare exception.

Generally the model is provider/kernel binding is only by DRIVER_ID and any variance beyond that should be fully negotiated as part of get_context.

**Comment by galpress** [comment]
> All the providers are expected to negotiate with the kernel if the right combination stuff is supported.
> 
> Will this upgraded provider work on old kernels? That would be the rare exception.

Probably, how old is old?

> Generally the model is provider/kernel binding is only by DRIVER_ID and any variance beyond that should be fully negotiated as part of get_context.

**Comment by jgunthorpe** [comment]
How Old? All prior kernels

**Comment by galpress** [comment]
Well, the provider is expected to work with prior kernels as well.
I don't think I understand what you mean?

**Comment by jgunthorpe** [comment]
Well, lets see a patch that doesn't use PCI_ID then maybe it will be clear

**Comment by galpress** [comment]
Done.
The pciid patch is still here because I had some issues without it, let's discuss it.

**Comment by jgunthorpe** [comment]
What are some issues without it?

**Comment by galpress** [comment]
The driver id binding didn't catch on one of the kernels I tested.

**Comment by galpress** [comment]
I assume you replied on the wrong thread:

> How did such an old kernel have support for this new EFA device?

An older kernel with backported 0xefa1 support.

**Comment by rleon** [comment]
> I assume you replied on the wrong thread:
> 
> > How did such an old kernel have support for this new EFA device?
> 
> An older kernel with backported 0xefa1 support.

Is this upstream kernel?
We rarely care about other cases.

**Comment by galpress** [comment]
It's not an upstream kernel, but specific distros might backport these patches to support new devices.
I get the upstream kernel point, but we can't have these new devices only working on latest upstream kernel..
Do you strongly object keeping the pciid single liner? I can't see a way we keep shipping our customers upstream rdma-core without it :\\.

**Comment by jgunthorpe** [comment]
I'd rather we not have forks of rdma-core for such a trivial thing. It seems we will have to wait a bit before going to the no new pci_id's.

**Comment by galpress** [comment]
Updated the kernel headers commit, let me know if anything else is needed.

**Comment by galpress** [comment]
Kind reminder.

**Comment by galpress** [comment]
Is there anything stopping this PR from being merged?

**Pull request merged**